src/demux/readers/eos/MongoActionReader.ts

```diff
@@ -0,0 +1,65 @@
+import { AbstractActionReader } from "../AbstractActionReader"
+import { MongoBlock } from "./MongoBlock"
+
+import { MongoClient } from "mongodb"
+
+/**
+ * Implementation of an ActionReader that polls a mongodb.
+ */
+export class MongoActionReader extends AbstractActionReader {
+  protected isInitialized: boolean
+  protected mongodb: any
+  constructor(
+    protected mongoEndpoint: string = "mongodb://127.0.0.1:27017",
+    public startAtBlock: number = 1,
+    public dbName: string = "EOS",
+    protected onlyIrreversible: boolean = false,
+    protected maxHistoryLength: number = 600,
+  ) {
+    super(startAtBlock, onlyIrreversible, maxHistoryLength)
+    this.mongoEndpoint = mongoEndpoint
+    this.mongodb = null
```

Comment:
This is the only line you need in the constructor, the other 3 lines are handled by the declaration above.